src/util/strencodings.cpp

```diff
@@ -46,14 +46,20 @@ bool IsHex(std::string_view str)
     return (str.size() > 0) && (str.size()%2 == 0);
 }
 
-bool IsHexNumber(std::string_view str)
+std::optional<std::string> TrySanitizeHexNumber(std::string_view input, int result_size)
 {
-    if (str.substr(0, 2) == "0x") str.remove_prefix(2);
-    for (char c : str) {
-        if (HexDigit(c) < 0) return false;
+    input = util::RemovePrefixView(input, "0x");
+    if (result_size < 0) result_size = input.size(); // negative result_size disables the size check
+    if (input.empty() || ((int)input.size() > result_size)) return std::nullopt;
```

Comment:
I don't particularly  like the alternative, but the developer notes state:
```
- Use a named cast or functional cast, not a C-Style cast. When casting
    between integer types, use functional casts such as `int(x)` or `int{x}`
    instead of `(int) x`. When casting between more complex types, use `static_cast`.
 ```

Reply:
Thanks, will change to `int{x}` in next force push.

Reply:
That might not work, but extracting might:
```C++
    auto final_size = (result_size < 0) ? input.size() : static_cast<size_t>(result_size);
```